components/CreateReplyRequestDialog/CreateReplyRequestDialog.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+
+const localStorage = typeof window === 'undefined' ? {} : window.localStorage;
+const formInitialState = {
+  visible: false,
+  disabled: true,
+  text: '',
+};
+
+const CREATE_REPLY_REQUEST = gql`
+  mutation CreateReplyRequestFromForm($articleId: String!, $reason: String!) {
+    CreateReplyRequest(articleId: $articleId, reason: $reason) {
+      id
+    }
+  }
+`;
+
+function SubmitButton({ articleId, text, disabled }) {
+  const [createReplyRequest] = useMutation(CREATE_REPLY_REQUEST, {
+    refetchQueries: ['LoadArticlePage'],
```

Comment:
Please clear the form (get the ref of `<form>` and invoke its [`reset()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reset) ) after the request completes.

Also, please set form state's `visible` to false after submission.

I think putting useMutation directly inside `CreateReplyRequestDialog` and avoid creating an isolate `SubmitButton` component can make things easier, and it is natural for a "Form" component to handle its submission.

Reply:
I tried to call "useMutation" in different functions of CreateReplyRequestDialog, however, it would complain about "Invalid hook call", and the page won't show up at all.

https://github.com/Stimim/rumors-site/commit/31179aa3e9ecb393c70e757e99b16f7f6c17a6dc

Reply:
Oh it's because hooks [can only be used in functional components](https://reactjs.org/docs/hooks-faq.html#should-i-use-hooks-classes-or-a-mix-of-both), so `CreateReplyRequestDialog` must be rewritten to functions in order to use `useMutation`.

Reply:
I am ok with current settings though, since it's working now